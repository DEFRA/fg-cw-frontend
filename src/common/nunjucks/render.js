import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
