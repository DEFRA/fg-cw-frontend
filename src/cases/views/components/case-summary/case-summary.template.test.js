import { describe, expect, test } from "vitest";
import { render } from "../../../../common/nunjucks/render.js";

describe("case-summary", () => {
  test("renders", () => {
    const component = render("case-summary", {
      case: {
        caseRef: "banana-123",
        status: "NEW",
        businessName: "Test Business",
        sbi: "SBI001",
        code: "frps-private-beta",
        scheme: "SFI",
        dateReceived: "2025-06-11T10:43:01.603Z",
      },
    });

    expect(component).toMatchSnapshot();
  });
});
