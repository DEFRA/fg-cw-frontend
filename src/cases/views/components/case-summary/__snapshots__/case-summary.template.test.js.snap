// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`case-summary > renders 1`] = `
"
<div class="govuk-!-margin-bottom-6 app-case-banner">
  <div class="govuk-width-container">
    <a href="/cases" class="govuk-back-link">Back to cases</a>

    <h1 class="govuk-heading-m">
      Valley Farm
    </h1>

    <div>
      <div class="app-case-row">
      </div>
    </div>
  </div>
</div>
"
`;
